src/overlays/actors/ovl_En_Fr/z_en_fr.c

```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+    this->actor.posRot.pos.y = sLogSpotToFromWater[this->actor.params].yDist + this->posLogSpot.y; // -80.0
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1AC34.s")
+    yaw = (s16)(sLogSpotToFromWater[this->actor.params].yaw * (32768.0f / M_PI)) + 32768;
 
+    this->actor.shape.rot.y = yaw;
+    this->actor.posRot.rot.y = yaw;
+    this->actor.speedXZ = 0.0f;
+    this->actor.velocity.y = 0.0f;
+    this->actor.gravity = 0.0f;
+}
+
+void EnFr_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnFr* this = THIS;
+
+    // 6 Actor Instances for frogs:
+    //     -  5 NPC actor instances for the frogs themselves
+    //     -  1 Prop actor instance set to where Link plays Ocarina, manages 5 frogs
+    // If (Prop Actor):
+    if (this->actor.params == 0) {
+        this->actor.destroy = NULL;
+        this->actor.draw = NULL;
+        this->actor.update = EnFr_UpdateIdle;
+        this->actor.flags = this->actor.flags &= ~0x11; // devsplz...
+
+        Actor_ChangeType(globalCtx, &globalCtx->actorCtx, &this->actor, ACTORTYPE_PROP);
+        this->actor.textId = 0x40AC;
+        this->actionFunc = EnFr_Idle;
+    } else {
+        // Translation: The argument is wrong!!
+        if ((this->actor.params >= 6) || (this->actor.params < 0)) {
+            osSyncPrintf(VT_COL(RED, WHITE));
+            osSyncPrintf("%s[%d] : 引数が間違っている！！(%d)\n", "../z_en_fr.c", 370, this->actor.params);
+            osSyncPrintf(VT_RST);
+            __assert("0", "../z_en_fr.c", 372);
+        }
+
+        this->objBankIndex = Object_GetIndex(&globalCtx->objectCtx, OBJECT_GAMEPLAY_FIELD_KEEP);
+
+        // Translation: There is no bank!!
+        if (this->objBankIndex < 0) {
+            Actor_Kill(&this->actor);
+            osSyncPrintf(VT_COL(RED, WHITE));
+            osSyncPrintf("%s[%d] : バンクが無いよ！！\n", "../z_en_fr.c", 380);
+            osSyncPrintf(VT_RST);
+            __assert("0", "../z_en_fr.c", 382);
+        }
+    }
+}
+
+void EnFr_DrawIdle(EnFr* this) {
+    // Draw only purple frog when not actively listening to ocarina
+    if (this->actor.params != 4) {
+        this->actor.draw = NULL;
+    } else {
+        this->actor.draw = EnFr_Draw;
+    }
+}
+
+void EnFr_DrawActive(EnFr* this) {
+    this->actor.draw = EnFr_Draw;
+}
+
+
+#ifdef NON_MATCHING
+void EnFr_Update(Actor* thisx, GlobalContext* globalCtx) {
+
+    EnFr* this = THIS;
+    LightInfo* lightInfo;
+    s32 frogindex;
+    s16 randomNumber;
+
+    // if (globalCtx) {}
+
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->objBankIndex)) {
+        this->actor.flags &= ~0x10;
+        frogindex = this->actor.params - 1;
+        sEnFrPointers.frogs[frogindex] = this;
+        Actor_ProcessInitChain(&this->actor, sInitChain);
+
+        // Skeleton for the frog
+        SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_0600B498, &D_06001534, &this->limbDrawTable,
+                         &this->transitionDrawTable, 24);
+        // Skeleton for the butterfly
```

Comment:
these comments will be uneeded when the skeleton vars are named, but you can leave it for now if you want